Check if a base image exists

mmachatschek opened the pull request.

Commit:
Add check for missing base image

Commit:
Add test for missing base image

Commit:
Update exception message

Commit:
Remove variable assignement as it is not necessary

howard commented on src/FINDOLOGIC/Export/XML/XMLItem.php:
```diff
@@ -102,9 +102,11 @@ private function buildImages(\DOMDocument $document)
-            /** @var Image $image */
-            foreach ($images as $image) {
-                $usergroupImagesElem->appendChild($image->getDomSubtree($document));
+            if ($this->validateImages($images)){
```
Space before `{` is missing.

mmachatschek replied:
Done

howard commented on src/FINDOLOGIC/Export/XML/XMLItem.php:
```diff
@@ -122,4 +124,22 @@ private function buildUsergroups(\DOMDocument $document)
+        $valid = false;
+
+        foreach ($images as $image) {
+            if($image->getType() === Image::TYPE_DEFAULT){
```
Space after `if` is missing.

mmachatschek replied:
Done

howard commented on src/FINDOLOGIC/Export/XML/XMLItem.php:
```diff
@@ -122,4 +124,22 @@ private function buildUsergroups(\DOMDocument $document)
+            throw new \FINDOLOGIC\Export\Data\BaseImageMissingException();
+        }
+
+        return $valid;
```
Is a return value actually necessary if an exception is thrown? This will always be `true`.

mmachatschek replied:
Done

howard commented on src/FINDOLOGIC/Export/XML/XMLItem.php:
```diff
@@ -122,4 +124,22 @@ private function buildUsergroups(\DOMDocument $document)
         return $usergroups;
     }
+
+    private function validateImages(array $images){
```
A doc block would be nice to document parameters and describe what validity actually means.

mmachatschek replied:
done

howard reviewed (changes requested):
Some things could be improved, but it does the right thing. Nice!

Commit:
Update whitespaces

Commit:
Update exception message

Commit:
Add docblock for validateImages method

Commit:
Import exception

Commit:
Remove obsolete return statement

Commit:
Add exception for missing images without usergroup

Commit:
Update check for images

Commit:
Add test for items with missing base image block and update existing test

mmachatschek commented:
@howard 
Added check for existing images without usergroup. If so, the normal check for an existing base image will apply. Furthermore I also added the check if there actually exist images, so the other test do not break because of the exception thrown when no images without usergroups exist.
